custom_components/elasticsearch/datastreams/index_template.json

```diff
@@ -117,6 +117,15 @@
                                 "platform": {
                                     "type": "keyword"
                                 },
+                                "unit_of_measure": {
+                                    "type": "keyword"
+                                },
+                                "class": {
+                                    "type": "keyword"
+                                },
+                                "labels": {
+                                    "type": "keyword"
+                                },
```

Comment:
The way this is currently written, an entity without a device will not have a `floor` assigned, as we are only recording this attribute within the `device` property below. It's also possible for an entity to be located in a different area (and therefore a different floor) than the device which owns it.

One common example is a thermostat with remote sensors. In my home, the thermostat is located within the dining room of the first floor, but the remote sensors are located in various areas which span the basement, ground floor, and upper floor.

I think it would make sense to record `floor` for both the `entity` and `device`, much like we do for `area` today.